graphql_compiler/compiler/emit_cypher.py

```diff
@@ -127,10 +148,64 @@ def _emit_with_clause_components(cypher_steps):
     return result
 
 
+def _emit_with_clause_components_for_current_fold_scope(current_fold_scope_cyphersteps):
+    """Emit the component strings for Cypher WITH clause for current fold."""
+    # At this point we should already have emitted all component strings from previous CypherSteps,
+    # whether or not they were in fold scopes or not.
+    result = []
+
+    vertex_names = {}
+    for cypher_step in current_fold_scope_cyphersteps:
+        fold_scope_location = cypher_step.as_block.location
+        full_vertex_name = fold_scope_location_full_path_name(fold_scope_location)
+        vertex_names[u'collect(' + full_vertex_name + ')'] = u'collected_' + full_vertex_name
```

Comment:
Maybe another good opportunity for a helper function: `collected_' + fold_scope_location_full_path_name(fold_scope_location)` exists both here and on line 135. That means that if we change the naming of this variable, we have to find both places in the code and e.g. modify two constants that happen to be set to the same literal value but are not otherwise related to each other.

Reply:
Good idea-- caught another example in expressions.py too